### ISSUE TITLE
PROMIS Physical Function Upper Extremity v2.0 Extension

### ISSUE THREAD

**Issue by ben-domingue** [ben-domingue/irw#525]
https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/IHNIRH

**Comment by KingArthur0205** [comment]
**PROMIS_Physical_Function_Short_Form_df**: 600 participants, 8 items, 5=Without any difficulty; 4=With a little difficulty; 3=With some difficulty; 2=With much difficulty; 1=Unable to do;
**PROMIS_Physical_Function_Upper_Extermity_df**: 600 participants, 50 items, 5=Without any difficulty; 
4=With a little difficulty; 3=With some difficulty; 2=With much difficulty; 1=Unable to do;
**Flexilevel_Scale_of_Shoulder_Function_df**:600 participants, 34 items,0=I can't do this;
1=Great difficulty; 2=Some difficulty; 3=Little difficulty; 4=No difficulty;

**Comment by KingArthur0205** [comment]
Data:
[PROMISPFUE_2.0E_Gershon_2019_PROMIS.csv](https://github.com/user-attachments/files/17530315/PROMISPFUE_2.0E_Gershon_2019_PROMIS.csv)

Code:
```
# Paper：https://osf.io/ad6b3/
# Data:https://pubmed.ncbi.nlm.nih.gov/34639633/

library(haven)
library(dplyr)
library(tidyr)
library(openxlsx)
library(readr)
library(readxl)
library(sas7bdat)

remove_na <- function(df) {
  df <- df[!(rowSums(is.na(df[, -which(names(df) %in% c("id"))])) == (ncol(df) -1)), ]
  return(df)
}

data_df <- read_csv("PROMIS_UE_data.csv")
data_df  <- data_df %>%
  rename(id = RID)

 
PROMIS_Physical_Function_Short_Form_df <- data_df |>
  select(PFA4:PFA11, id)
PROMIS_Physical_Function_Short_Form_df <- remove_na(PROMIS_Physical_Function_Short_Form_df )
PROMIS_Physical_Function_Short_Form_df <- PROMIS_Physical_Function_Short_Form_df  %>%
  pivot_longer(cols = -c(id), names_to = "item", values_to = "resp")

PROMIS_Physical_Function_Upper_Extermity_df <- data_df |>
  select(QSD6, PFA14:PFM18, id)
PROMIS_Physical_Function_Upper_Extermity_df  <- remove_na(PROMIS_Physical_Function_Upper_Extermity_df)
PROMIS_Physical_Function_Upper_Extermity_df <- PROMIS_Physical_Function_Upper_Extermity_df %>%
  pivot_longer(cols = -c(id), names_to = "item", values_to = "resp")

Flexilevel_Scale_of_Shoulder_Function_df <- data_df |>
  select(QFLEXB1:QFLEXH15, id)
Flexilevel_Scale_of_Shoulder_Function_df  <- remove_na(Flexilevel_Scale_of_Shoulder_Function_df)
Flexilevel_Scale_of_Shoulder_Function_df <- Flexilevel_Scale_of_Shoulder_Function_df %>%
  pivot_longer(cols = -c(id), names_to = "item", values_to = "resp")
Flexilevel_Scale_of_Shoulder_Function_df$resp <- Flexilevel_Scale_of_Shoulder_Function_df$resp + 1

df <- rbind(PROMIS_Physical_Function_Short_Form_df, PROMIS_Physical_Function_Upper_Extermity_df)
df<- rbind(df, Flexilevel_Scale_of_Shoulder_Function_df)
save(Flexilevel_Scale_of_Shoulder_Function_df, file="PROMISPFUE_2.0E_Gershon_2019_PROMIS.Rdata")
write.csv(Flexilevel_Scale_of_Shoulder_Function_df, "PROMISPFUE_2.0E_Gershon_2019_PROMIS.csv", row.names=FALSE)

```

**Comment by KingArthur0205** [comment]
PR for issue: https://github.com/ben-domingue/irw/pull/642

**Comment by KingArthur0205** [comment]
Added to the dictionary spreadsheet and currently marked yellow

**Comment by ben-domingue** [comment]
@KingArthur0205 you noted "Although these are different scales, they all measure physical functioning(arms/shoulders) of participants. I decided to merge them into one final dataset. I tweaked the scale of Flexilevel_Scale_of_Shoulder_Function_df to 1-5 to align with the other 2 scales." this isn't a problem here but don't worry about this in general. the irt models for polytomous items can deal with different numbers of categories flexibly
